### ISSUE TITLE
emlinefit hangs on 82406/20211118/coadd-5-82406-thru20211118.fits

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1985]
@araichoor the following emlinefit command hangs (no errors, no crash, it just gets stuck doing something):

```
desi_emlinefit_afterburner \
  --coadd /global/cfs/cdirs/desi/spectro/redux/iron/tiles/cumulative/82406/20211118/coadd-5-82406-thru20211118.fits \
  --redrock /global/cfs/cdirs/desi/spectro/redux/iron/tiles/cumulative/82406/20211118/redrock-5-82406-thru20211118.fits \
  --output $SCRATCH/emline-5-82406-thru20211118.fits
```
The same occurs for spectrograph 6, but the other petals from that tile work fine.

This causes this ztile job to timeout with those two missing emline files.  Please check and see if you can implement a fix that wouldn't change the answer for anything that has already been run.

**Comment by araichoor** [comment]
investigation report:

I guess the reason is because it s a tile with insane ebv.. it s a `specialbackup` test tile at galactic `b=-2`...
in short:
- in emlinefit, the flux is corrected for Galactic extinction, here:
https://github.com/desihub/desispec/blob/e8c541d463f3726a1bf7dba633c6306c4358683a/py/desispec/io/emlinefit.py#L239-L246

- some fibers have flux > 1e4, so it shoots up to 1e6-1e7 after correction, likely being the reason for the code hanging.

a possibility is to simply remove that tile :)
if we want a code fix for that, nulling the ivar for insane fluxes works.
i.e. add at the end of this loop the following lines
```
                # AR nulling ivar for insane flux
                sel = tmpfl[i, :] > 1e7
                if sel.sum() > 0:                                                                                                                                                                   
                    tmpiv[i, sel] = 0
```
we could add a condition to perform this test only if `TILEID=82406`, in which case we would be 100% certain that we do not change any computation for other tiles.

few further infos/details:

here is the position of this tile in Galactic coordinates (displaying all iron tiles):
![tmp-iron-gal](https://user-images.githubusercontent.com/61986357/215028865-28eb8342-f57e-4bd9-b827-fd815fbca966.png)

(and here the tile design: https://desi.lbl.gov/trac/wiki/SurveyOps/TileDesigns#a82399-82411-BACKUPprogramtests)

I looked at the first row where it hangs, i.e. `TARGETID=2305843012414221480` (index 103; it s a Gaia/G2 star of mag ~15).
note that it also hangs for other rows.

the code hangs here when trying to fit the OII line in the `3687.2 A < lambda_obs < 3769.6 A` region:
https://github.com/desihub/desispec/blob/e8c541d463f3726a1bf7dba633c6306c4358683a/py/desispec/emlinefit.py#L218-L227
here is the flux for that spectrum (with highlighting in light grey the region used for the fit):
![tmp-2305843012414221480](https://user-images.githubusercontent.com/61986357/215029594-81ca24eb-d12a-44a7-8c86-ef5e0dd6e7a7.png)
the flux there can be of ~1e4 1e-17 ergs/s/cm2/A; and, when corrected with the super-high extinction factor, it shoots up to ~1e6-1e7 ergs/s/cm2/A.

**Comment by moustakas** [comment]
I agree it's a pretty extreme position in the sky, but---FWIW---FastSpecFit doesn't crash on this tile, so I'm not sure throwing it away completely is the right choice.

**Comment by sbailey** [comment]
Accepting 2 missing afterburner files is indeed better than discarding the entire tile, even if it was a backup program test tile :)

Although the flux numbers are big, they don't seem big enough to run into overflow issues that would actually cause a fit failure.  I think there is something else making this a pathological case.

**Comment by sbailey** [comment]
fixed by PR #2195
